File: Iterator/Python/ListaNumerosIterador.py
from Iterador import *

class ListaNumerosIterado(Iterador):

    def __init__(self, numeros):
        self._numeros = numeros
        self._posicion = 0
    
    def siguiente(self):
        valor = self._numeros[self._posicion]
        self._posicion += 1
        return valor
    
    def tieneSiguiente(self):
        if  self._posicion < len(self._numeros):
            return True
        else:
            return False
        

    

File: Iterator/Java/ListaNumerosIterador.java
public class ListaNumerosIterador implements Iterador{
    private int numeros[];
    private int posicion;

    ListaNumerosIterador(int num[])
    {
        this.numeros = num;
        this.posicion = 0;
    }

    @Override
    public Object siguiente() {
        return numeros[posicion++];
    }

    @Override
    public boolean tieneSiguiente() {
        if (this.posicion < numeros.length )
        {
            return true;
        }
        else{
            return false;
        }
    }
}
